lib/bolt/applicator.rb

```diff
@@ -188,5 +192,41 @@ def apply(args, apply_body, scope)
       end
       r
     end
+
+    def plugins
+      @plugin_tarball.value or raise Bolt::Error.new("Failed to pack module plugins: #{@plugin_tarball.reason}", 'bolt/plugin-error')
+    end
+
+    def build_plugin_tarball
+      start_time = Time.now
+      plugins = Tempfile.new('plugins')
+      sio = StringIO.new
+      output = Minitar::Output.new(Zlib::GzipWriter.new(sio))
```

Comment:
This could get more complicated with Puppet 6, which appears to stop shipping minitar.